lib/daru/dataframe.rb

```diff
@@ -1419,6 +1422,29 @@ def report_building(b) # :nodoc: #
         end
       end
     end
+    #custom function for dataframe summary
+    def g_summary()
+       frame_summary = "= \t\n"
+       frame_summary << "\tNumber of rows: #{nrows}\n"
+       @raw = self
```

Comment:
Why not use `self` directly? Using instance variables like this is not acceptable since it stays for absolutely no reason in the object and consumes resources. IVs should not be initialized unless you absolutely must.